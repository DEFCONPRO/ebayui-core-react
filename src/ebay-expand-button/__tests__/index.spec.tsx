import React from 'react'
import { fireEvent, render } from '@testing-library/react';
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayExpandButton } from '../../index'

initStoryshots({
    config: ({ configure }) => configure(() => {require('./index.stories')}, module)
})

describe('<EbayExpandButton>', () => {
    describe('on button click', () => {
        let spy: any
        beforeEach(() => {
            spy = jest.fn()
            spy.mockReset()
        })
        it('should fire onClick event', () => {
            const wrapper = render(
                <EbayExpandButton onClick={spy} />
            )
            fireEvent.click(wrapper.getByRole('button'))

            expect(spy).toBeCalled()
        })
        it('should fire onExpand event', () => {
            const wrapper = render(
                <EbayExpandButton onExpand={spy} />
            )
            fireEvent.click(wrapper.getByRole('button'))

            expect(spy).toBeCalled()
        })
        it('should fire onCollapse event', () => {
            const wrapper = render(
                <EbayExpandButton onCollapse={spy} />
            )
            const button = wrapper.getByRole('button');
            fireEvent.click(button)
            fireEvent.click(button)

            expect(spy).toBeCalledTimes(1)
        })
    })
})
