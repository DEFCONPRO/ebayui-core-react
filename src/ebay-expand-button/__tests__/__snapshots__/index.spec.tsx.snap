// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-expand-button Borderless 1`] = `
<span
  class="listbox-button"
>
  <button
    aria-expanded="false"
    class="expand-btn expand-btn--borderless"
  >
    <span
      class="expand-btn__cell"
    >
      <span
        class="expand-btn__text"
      >
        Borderless Button
      </span>
      <svg
        aria-hidden="true"
        class="icon icon--dropdown"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</span>
`;

exports[`Storyshots ebay-expand-button Default 1`] = `
<button
  aria-expanded="false"
  class="expand-btn"
>
  <span
    class="expand-btn__cell"
  >
    <span
      class="expand-btn__text"
    >
      Expand!
    </span>
    <svg
      aria-hidden="true"
      class="icon icon--dropdown"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-dropdown"
      />
    </svg>
  </span>
</button>
`;

exports[`Storyshots ebay-expand-button Disabled 1`] = `
<button
  aria-expanded="false"
  class="expand-btn"
  disabled=""
>
  <span
    class="expand-btn__cell"
  >
    <span
      class="expand-btn__text"
    >
      Can not touch
    </span>
    <svg
      aria-hidden="true"
      class="icon icon--dropdown"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-dropdown"
      />
    </svg>
  </span>
</button>
`;

exports[`Storyshots ebay-expand-button Html Label 1`] = `
<button
  aria-expanded="false"
  class="expand-btn"
>
  <span
    class="expand-btn__cell"
  >
    <span
      class="expand-btn__text"
    >
      <h3>
        Header
      </h3>
    </span>
    <svg
      aria-hidden="true"
      class="icon icon--dropdown"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-dropdown"
      />
    </svg>
  </span>
</button>
`;

exports[`Storyshots ebay-expand-button No text 1`] = `
<p>
  <button
    aria-expanded="false"
    aria-label="Menu"
    class="expand-btn expand-btn--icon-only"
  >
    <span
      class="expand-btn__cell"
    >
      <svg
        aria-hidden="true"
        class="icon icon--dropdown"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
</p>
`;
